### PR TITLE
Bump compat for Introduction to OptimizationProblems.jl tutorial

### PR DESCRIPTION
**Pull request by tmigot**
**New tutorial**
**Updates to tutorial**

- [ ] Tutorial name:
- [ ] Confirm that either:
  - There is an issue for this tutorial:
    - Close https://github.com/JuliaSmoothOptimizers/JSOTutorials.jl/issues/142
  - It was discussed in some other manner.
- [ ] Confirm that you have used a tool such as [Grammarly] to verify that the text is correct.
- Wait for the workflow, look at the [preview], and confirm:
  - [ ] The preview has your tutorial listed.
  - [ ] The preview looks like what you want.

[preview]: https://jso.dev/JSOTutorials.jl/
[Grammarly]: https://www.grammarly.com

### CODE REVIEW

**Review by farhadrclass — COMMENTED** [review]
Looks good,
I suggest we add a note section mentioning that from new version. we do not accept Val(T) so people who already have the code and update can refer to it and also we need to double check the SolverBenchmark tutorial

File: tutorials/introduction-to-optimizationproblems/index.jmd [hunk 1]
```diff
@@ -64,7 +64,7 @@ nlp_120 = OptimizationProblems.ADNLPProblems.woods(n=120)
 ```
 One of the advantages of these problems is that they are type-stable. Indeed, one can specify the output type with the keyword `type` as follows.
```
Comment (tmigot) [inline tutorials/introduction-to-optimizationproblems/index.jmd:65]:
```suggestion
One of the advantages of these problems is that they are type-stable. Indeed, one can specify the output type with the keyword `type` as follows. Note that in version < 0.8 the argument was `type=Val(DataType)`.
```

Comment (tmigot) [reply tutorials/introduction-to-optimizationproblems/index.jmd:65]: @farhadrclass something like this?

Comment (farhadrclass) [reply tutorials/introduction-to-optimizationproblems/index.jmd:65]: Looks good

**Review by farhadrclass — APPROVED** [review]

**Pull request merged**